packages/adapter-cloudflare/src/worker.js

```diff
@@ -36,26 +37,30 @@ const worker = {
 				}
 			});
 		} else {
-			// prerendered pages and index.html files
-			pathname = pathname.replace(/\/$/, '') || '/';
-
-			let file = pathname.substring(1);
+			// prerendered pages and /static files
 
 			try {
-				file = decodeURIComponent(file);
-			} catch (err) {
-				// ignore
+				pathname = decodeURIComponent(pathname);
+			} catch {
+				// ignore invalid URI
 			}
 
-			if (
-				manifest.assets.has(file) ||
-				manifest.assets.has(file + '/index.html') ||
-				prerendered.has(pathname)
-			) {
+			const stripped_pathname = pathname.replace(/\/$/, '') || '';
+
+			let is_asset = false;
+			const filename = stripped_pathname.substring(1);
+			if (filename) {
+				is_asset = manifest.assets.has(filename) || manifest.assets.has(filename + '/index.html');
+			}
```

Comment:
Could there theoretically be an asset named `index.html`? What was the reason for changing this?

Reply:
> Could there theoretically be an asset named index.html?
If a dev decides to include one in `/static` I suppose. It helps if someone navigates to that route.
This check was already there previously, but I suppose it's good to have it since the worker has to handle all file routing.

> What was the reason for changing this?
I thought it would be a small optimisation, although I'm unsure if it has an impact.
If the url is `https://test.com/`, then the resulting filename would be `''`. This skips the manifest check.